qtdialogs/DlgSetupManager.py

```diff
@@ -0,0 +1,1197 @@
+##############################################################################
+#                                                                            #
+#  Copyright (C) 2025, goatpig                                               #
+#  Distributed under the MIT license                                         #
+#  See LICENSE-MIT or https://opensource.org/licenses/MIT                    #
+#                                                                            #
+##############################################################################
+
+import os
+from qtpy import QtCore, QtWidgets
+from armoryengine.ArmoryUtils import BTC_HOME_DIR, ARMORY_DB_DIR, ARMORY_HOME_DIR, \
+   OS_MACOSX, OS_WINDOWS, CLI_OPTIONS, getBridgeArgList, LOGINFO, LOGEXCEPT
+from armoryengine.Settings import TheSettings
+from armoryengine.CppBridge import TheBridge
+from armoryengine.BDM import TheBDM
+
+from qtdialogs.ArmoryDialog import ArmoryDialog
+from armorycolors import htmlColor
+from qtdialogs.DlgMigrateWallet import DlgMigrateWallet
+from qtdialogs.DlgUnlockWallet import UnlockWalletHandler
+from qtdialogs.qtdefines import (
+   UI_DIALOG_SPACING, UI_FRAME_MARGIN, UI_FRAME_PADDING, UI_BUTTON_SPACING, UI_GRID_SPACING,
+   UI_STYLE_BUTTON_STANDARD, UI_STYLE_BUTTON_DIALOG, UI_STYLE_INPUT, UI_STYLE_COMBOBOX,
+   UI_STYLE_FRAME, UI_STYLE_TREEWIDGET, apply_dialog_base_style,
+   createStyledLabel, createButtonLayout, createInputField, createStyledButton, createStyledCombo,
+   ComboBoxStyle
+)
+
+# --- Dialog-specific constants ---
+MINIMUM_DIALOG_WIDTH = 850
+MINIMUM_DIALOG_HEIGHT = 500
+
+###############################################################################
+class DlgSetupManager(ArmoryDialog):
+   def __init__(self, parent=None, main=None, testing=False):
+      """Initialize the setup manager dialog."""
+      super().__init__(parent)
+      
+      self.initMemberVariables(main)
+      self.setupDialogProperties()
+      self.initTabs()
+      self.setupMainLayout()
+      self.loadSettings(testing)
+      self.connectSignals()
+      self.loadWalletList()
+
+      # Display dialog unless in testing mode
+      if not testing:
+         self.show()
+
+   def initMemberVariables(self, main):
+      """Initialize all member variables with default values."""
+      self.main = main
+      self.comboStyle = ComboBoxStyle()
+      self.bitcoinCoreProcess = None
+      
+      # UI component references
+      self.tabWidget = None
+      self.walletTab = None
+      self.coreTab = None
+      self.databaseTab = None
+      self.acceptButton = None
+      self.cancelButton = None
+      
+      # Widget references for settings
+      self.satoshiHomePath = None
```

Comment:
this is too hard to read.

break it down into relevant feature groups, implement the sanity checks there. It's fine to have a flat implementation for GUI, but the underling data should be isolated where possible.